packages/react-api/src/hooks/useCartoLayerProps.js

```diff
@@ -1,19 +1,29 @@
 import { DataFilterExtension } from '@deck.gl/extensions';
 import { _buildFeatureFilter } from '@carto/react-core';
-import useViewportFeatures from './useViewportFeatures';
 import { MAP_TYPES, API_VERSIONS } from '@deck.gl/carto';
+import { useSelector } from 'react-redux';
+import useGeoJsonFeatures from './useGeoJsonFeatures';
+import useTilesetFeatures from './useTilesetFeatures';
 
 export default function useCartoLayerProps({
   source,
   uniqueIdProperty,
   viewportFeatures = true,
-  viewporFeaturesDebounceTimeout = 500
+  viewporFeaturesDebounceTimeout = 250
 }) {
-  const [onViewportLoad, onDataLoad, fetch] = useViewportFeatures(
-    source,
+  const viewport = useSelector((state) => state.carto.viewport);
```

Comment:
Why have you pushed up this viewport selection and not left it inside the 'viewportFeatures' hooks?

Reply:
Because each time 'viewportFeatures' hooks changes, useCartoLayerProps changes, if the same change happens in two different hook that are children of the a certain hook, it's better to have that change listener (in this case the selector) in the parent hook. Also, it's a improvement to make `useGeoJsonFeatures` and `useTilesetFeatures` independent from viewport.

Reply:
I'm curious about independent from viewport in the future... you will tell us... 😄

Reply:
I hope to be able to tackle that soon! Many steps have to be taken before 😄